packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -554,6 +554,29 @@ HostPlatform getCurrentHostPlatform() {
   return HostPlatform.linux_x64;
 }
 
+/// Convert [buildInfo] to a structued string encoded structure appropriate for
+/// usage as environment variables or to embed in other scripts.
+///
+/// Fields that are `null` are excluded from this configration.
+Map<String, String> toEnvironmentConfig(BuildInfo buildInfo) {
```

Comment:
Why is this freestanding instead of a method on `BuildInfo`? (Especially since it's named like `toString`, which is a member.)

Reply:
No reason, I'll update it